odoo/tools/mail.py

```diff
@@ -2,255 +2,116 @@
-        el_id = el.get('id', '') or ''
-
-        # gmail or yahoo // # outlook, html // # msoffice
-        if 'gmail_extra' in el_class or \
-                'divRplyFwdMsg' in el_id or \
-                ('SkyDrivePlaceholder' in el_class or 'SkyDrivePlaceholder' in el_class):
-            el.set('data-o-mail-quote', '1')
-            if el.getparent() is not None:
-                el.getparent().set('data-o-mail-quote-container', '1')
-
-        if (el.tag == 'hr' and ('stopSpelling' in el_class or 'stopSpelling' in el_id)) or \
-           'yahoo_quoted' in el_class:
-            # Quote all elements after this one
-            el.set('data-o-mail-quote', '1')
-            for sibling in el.itersiblings(preceding=False):
-                sibling.set('data-o-mail-quote', '1')
-
-        # html signature (-- <br />blah)
-        signature_begin = re.compile(r"((?:(?:^|\n)[-]{2}[\s]?$))")
-        if el.text and el.find('br') is not None and re.search(signature_begin, el.text):
-            el.set('data-o-mail-quote', '1')
-            if el.getparent() is not None:
-                el.getparent().set('data-o-mail-quote-container', '1')
-
-        # text-based quotes (>, >>) and signatures (-- Signature)
-        text_complete_regex = re.compile(r"((?:\n[>]+[^\n\r]*)+|(?:(?:^|\n)[-]{2}[\s]?[\r\n]{1,2}[\s\S]+))")
-        if not el.get('data-o-mail-quote'):
-            _tag_matching_regex_in_text(text_complete_regex, el, 'span', {'data-o-mail-quote': '1'})
-
-        if el.tag == 'blockquote':
-            # remove single node
-            el.set('data-o-mail-quote-node', '1')
-            el.set('data-o-mail-quote', '1')
-        if el.getparent() is not None and (el.getparent().get('data-o-mail-quote') or el.getparent().get('data-o-mail-quote-container')) and not el.getparent().get('data-o-mail-quote-node'):
-            el.set('data-o-mail-quote', '1')
-
-    def strip_class(self, el):
-        if el.attrib.get('class'):
-            del el.attrib['class']
-
-    def parse_style(self, el):
-        attributes = el.attrib
-        styling = attributes.get('style')
-        if styling:
-            valid_styles = collections.OrderedDict()
-            styles = self._style_re.findall(styling)
-            for style in styles:
-                if style[0].lower() in self._style_whitelist:
-                    valid_styles[style[0].lower()] = style[1]
-            if valid_styles:
-                el.attrib['style'] = '; '.join('%s:%s' % (key, val) for (key, val) in valid_styles.items())
+
+def tag_quote(el):
+    """Quote the email content."""
+    def _create_new_node(tag, text, tail=None, attrs=None):
+        new_node = etree.Element(tag)
+        new_node.text = text
+        new_node.tail = tail
+        if attrs:
+            for key, val in attrs.items():
+                new_node.set(key, val)
+        return new_node
+
+    def _tag_matching_regex_in_text(regex, node, tag='span', attrs=None):
```

Comment:
Are those fully copy-pasted or not really?

If they are, would be nice to split in a preparatory commit to avoid re-reading while reviewing.